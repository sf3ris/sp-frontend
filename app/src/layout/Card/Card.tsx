import React from 'react';

interface ICardProps {
    title?: string;
}

const Card : React.FC<ICardProps> = props => {

    return (

        <article>
            <h3>{props.title}</h3>
        </article>

    )

}

export default Card;