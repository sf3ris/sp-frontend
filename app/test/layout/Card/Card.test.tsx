import React from 'react';
import Card from '../../../src/layout/Card/Card';
import { shallow } from 'enzyme';

describe('Card Layout Component', () => {

    it('Should match snapshot', () => {

        const wrapper = shallow( <Card title="card-title"/> );

        expect(wrapper.debug()).toMatchSnapshot();

    });

    it('Should containt an h3 tag with title inside', () => {

        const wrapper = shallow( <Card title="card-title"/> );

        const h3 = wrapper.find('h3');

        expect(h3.length).toBe(1);

        expect(h3.text()).toBe('card-title');

    });
    
})
