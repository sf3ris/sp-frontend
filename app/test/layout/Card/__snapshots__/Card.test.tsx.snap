// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Layout Component Should match snapshot 1`] = `
"<article>
  <h3>
    card-title
  </h3>
</article>"
`;
